exercises/practice/two-fer/.approaches/introduction.md

```diff
@@ -0,0 +1,33 @@
+# Introduction
+
+## Default Paramater
+
+The key to this exercise is the optional paramater in the method definition.
+Using a default value there, rather than a conditional, makes the code much more readable, simple, and idiomatic.
+This is a pretty standard solution.
+
+```ruby
+module TwoFer
+  def self.two_fer(name = 'you')
+    "One for #{name}, one for me."
+  end
+end
+```
+
+## Method definition
+
+You can use `class` instead of `module` and use metaclasses instead of `self`, `module_function`, or extend `self`.
```

Comment:
```suggestion
You may use `class` instead of `module` and use metaclasses instead of `self`, `module_function`, or extend `self`.
```

Reply:
For a practice exercise, perhaps.  This is not a concept exercise, so I am unsure we have to be as careful about "prior knowledge" since we may assume they are coming from other languages.

Reply:
We're assuming that people are coming from other languages when they do concept exercises, as well.

I'm not worried about mentioning things that haven't been introduced, I think that's a given in the approaches documents. What I'm wondering is whether we should be a bit more verbose or explanatory, since this is one of the very first practice exercises, and people might be coming from languages that don't have these concepts.

Reply:
That's definitely fair.  I think in their research of what a module is, they should have encountered the extend/include as that is pretty core to what modules are used for.  But linking to a resource about module and class and what the differences are would probably be welcome, especially early on.

Do we expect people to go through the concept exercises before they go to practice mode, if they are really new to the language?  So in the case of "new to language" rather than "practicing" would we expect this information be covered already?

Reply:
This would be unlocked to practice with almost as soon as you get started with the concept exercises, so I think it could go either way.

My gut feeling here is that it's _good enough_ for now, and that we can take a second pass once the whole approaches thing is in production. We'll probably find out soon enough if people get confused.